S3/FileLists.py

```diff
@@ -60,34 +60,37 @@ def handle_exclude_include_walk_dir(root, dirname):
     Should this root/dirname directory be excluded? (otherwise included by default)
     Exclude dir matches in the current directory
     This prevents us from recursing down trees we know we want to ignore
-    return True for including, and False for excluding
+    return True for excluding, and False for including
     '''
     cfg = Config()
+    # python versions end their patterns (from globs) differently, test for different styles
+    directory_patterns = [u'\\/$', u'\\/\\Z(?ms)', u'/)$', u'/)\\Z']
```

Comment:
it is a micro optimisation, but you can put the new pattern before in the list as it should be the one that will match most with recent versions.